File: backend/src/main/java/com/apocalypse/thefall/entity/instance/ArmorInstance.java
package com.apocalypse.thefall.entity.instance;

import com.apocalypse.thefall.entity.item.enums.EquippedSlot;
import jakarta.persistence.*;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.SuperBuilder;
import org.hibernate.annotations.JdbcType;
import org.hibernate.dialect.PostgreSQLEnumJdbcType;

@Entity
@Getter
@Setter
@Table(name = "armor_instance")
@DiscriminatorValue("ARMOR")
@NoArgsConstructor
@SuperBuilder
public class ArmorInstance extends ItemInstance {
    @Enumerated(EnumType.STRING)
    @Column(name = "equipped_slot")
    @JdbcType(PostgreSQLEnumJdbcType.class)
    @Builder.Default
    private EquippedSlot equippedSlot = null;
}

File: backend/src/main/java/com/apocalypse/thefall/entity/instance/ArmorInstance.kt
package com.apocalypse.thefall.entity.instance

// ArmorInstance.kt
import com.apocalypse.thefall.entity.item.enums.EquippedSlot
import jakarta.persistence.*
import org.hibernate.annotations.JdbcType
import org.hibernate.dialect.PostgreSQLEnumJdbcType

@Entity
@Table(name = "armor_instance")
@DiscriminatorValue("ARMOR")
open class ArmorInstance : ItemInstance() {

    @Enumerated(EnumType.STRING)
    @Column(name = "equipped_slot")
    @JdbcType(PostgreSQLEnumJdbcType::class)
    open var equippedSlot: EquippedSlot? = null
}
